extensions/Maps/includes/Maps_Geocoder.php

```diff
@@ -111,7 +111,21 @@ public function hasAlias( $alias ) {
 	 * @return array or false
 	 */
 	public function geocode( $address ) {
-		$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+
+		// MAIN-6654: cache requests to this API because it has daily limits
+		global $wgMemc;
+		$key = wfMemcKey( "geocoder", $address );
+		$response = $wgMemc->get( $key );
+		if ( empty($response) ) {
+			$response = ExternalHttp::get( $this->getRequestUrl( $address ) ); # Wikia change
+			// MAIN-6654: make sure we do not return or store a bad response because it causes errors later
+			if ( substr( $response, "OVER_QUERY_LIMIT" ) !== false ) {
```

Comment:
This is not how ```substr``` is supposed to be used - I think you are looking to use ```strpos```.
It could not work with ```substr``` being used - so how was it tested? (It should be tested by mocking the response from  the API.)
@sqreek: @owend is correct - the caching layer that was already provided wasn't shared between requests.